orttraining/orttraining/eager/ort_aten.cpp

```diff
@@ -1204,6 +1247,257 @@ at::Tensor squeeze(
     tensor_options);
 }
 
+// aten::add.out(Tensor self, Tensor other, *, Scalar alpha=1, Tensor(a!) out) -> Tensor(a!)
+at::Tensor& add_out(
+  const at::Tensor& self,
+  const at::Tensor& other,
+  // *,
+  const at::Scalar& alpha,
+  at::Tensor& out) {
+  ORT_LOG_FN(self, other, alpha, out);
+
+  auto promoted_type = PromoteScalarTypesWithCategory({other.scalar_type(),self.scalar_type()}, {alpha.type()});
+
+  if (
+    !IsSupportedType(alpha, {at::kBFloat16,at::kByte,at::kDouble,at::kFloat,at::kHalf,at::kInt,at::kLong,at::kShort}) ||
```

Comment:
nit: consider handling all of these errors by assigning the list of supported types once and reusing (see armax_out impl)